pkg/flink/handler.go

```diff
@@ -135,6 +136,25 @@ func (flinkResourceHandler) BuildIdentityResource(ctx context.Context, taskCtx p
 	}, nil
 }
 
+func (h flinkResourceHandler) OnAbort(ctx context.Context, tCtx pluginsCore.TaskExecutionContext, resource client.Object) (behavior k8s.AbortBehavior, err error) {
+	flinkCluster := resource.(*flinkOp.FlinkCluster)
+
+	if flinkCluster.Status.Components.Job != nil && flinkCluster.Status.Components.Job.State == flinkOp.JobStatePending {
+		return k8s.AbortBehaviorDeleteDefaultResource(), nil
+	}
+
+	annotationPatch, err := NewAnnotationPatch(flinkOp.ControlAnnotation, flinkOp.ControlNameJobCancel)
+	var abortBehavior k8s.AbortBehavior
+
+	if err == nil {
+		abortBehavior = k8s.AbortBehaviorPatchDefaultResource(
+			k8s.PatchResourceOperation{Patch: annotationPatch},
+			false)
```

Comment:
Note: the "false" flag here means: if applying the patch returns and error, do *not* delete the underlying resource. This seems like the right approach if, for example, a termination request is sent after the job has completed; but if it fails for a different reason maybe we want to delete it just to clean up.